code/modules/surgery/ear_surgery.dm

```diff
@@ -0,0 +1,75 @@
+//Head surgery to fix the ears organ
+/datum/surgery/ear_surgery
+	name = "Ear surgery"
+	requires_bodypart_type = NONE
+	organ_to_manipulate = ORGAN_SLOT_EARS
+	possible_locs = list(BODY_ZONE_HEAD)
+	steps = list(
+		/datum/surgery_step/incise,
+		/datum/surgery_step/retract_skin,
+		/datum/surgery_step/saw,
+		/datum/surgery_step/clamp_bleeders,
+		/datum/surgery_step/fix_ears,
+		/datum/surgery_step/close,
+	)
+
+//fix ears
+/datum/surgery_step/fix_ears
+	name = "fix ears (hemostat)"
+	implements = list(
+		TOOL_HEMOSTAT = 100,
+		TOOL_SCREWDRIVER = 45,
+		/obj/item/pen = 25)
+	time = 64
+
+/datum/surgery/ear_surgery/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/internal/ears/target_ears = target.getorganslot(ORGAN_SLOT_EARS)
+	if(!target_ears)
+		return FALSE
+	return TRUE
+
+/datum/surgery_step/fix_ears/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(
+		user,
+		target,
+		span_notice("You begin to fix [target]'s ears..."),
+		span_notice("[user] begins to fix [target]'s ears."),
+		span_notice("[user] begins to perform surgery on [target]'s ears."),
+	)
+	display_pain(target, "You feel a dizzying pain in your head!")
+
+/datum/surgery_step/fix_ears/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
+	var/obj/item/organ/internal/ears/target_ears = target.getorganslot(ORGAN_SLOT_EARS)
+	display_results(
+		user,
+		target,
+		span_notice("You succeed in fixing [target]'s ears."),
+		span_notice("[user] successfully fixes [target]'s ears!"),
+		span_notice("[user] completes the surgery on [target]'s ears."),
+	)
+	display_pain(target, "Your head swims, but it seems like you can feel your hearing coming back!")
+	target_ears.deaf = (20) //deafness works off ticks, so this should work out to about 30-40s
+	target_ears.setOrganDamage(0)
```

Comment:
Why not use ``adjustEarDamage`` here?

Reply:
adjustEarDamage is incremental and, as far as I'm aware, if I pick a value for it except something like the current damage on the ears, it might not play nice with larger or smaller amounts of damage. If they've got damage and deafness and I add extra deafness after removing the damage, the damage removal loses a lot of its point as they still wait for the original deafness to clear. The point is to clear all the damage but leave some deafness which then naturally clears over the next 40s or so. For large amounts of ear damage, depending on RNG, this seems to be less than the amounts you might stack up from the actual deafness code in testing:

`	if((organ_flags & ORGAN_FAILING))
		deaf = max(deaf, 1) // if we're failing we always have at least 1 deaf stack (and thus deafness)
	else // only clear deaf stacks if we're not failing
		deaf = max(deaf - (0.5 * delta_time), 0)
		if((damage > low_threshold) && DT_PROB(damage / 60, delta_time))
			adjustEarDamage(0, 4)
			SEND_SOUND(owner, sound('sound/weapons/flash_ring.ogg'))`